dhcp/config.go

```diff
@@ -118,3 +118,26 @@ func (c *Config) SetDNSServers(ips []net.IP) {
 	}
 	c.opts.SetOption(dhcp4.OptionDomainServer, b)
 }
+
+// SetOpt43SubOpt sets an option 43 sub-option. If option 43 is already set, the sub-option is appended.
+func (c *Config) SetOpt43SubOpt(subOpt dhcp4.Option, s string) {
+	if s == "" {
+		return
+	}
+	// get current option 43
+	cur, ok := c.opts.GetOption(dhcp4.OptionVendorSpecific)
+	if !ok {
+		c := make(dhcp4.OptionMap, 0)
+		cur = c.Serialize()
+	}
+	// deserialize it into a dhcp4.OptionMap
+	n := make(dhcp4.OptionMap, 0)
+	if err := n.Deserialize(cur, nil); err != nil {
+		dhcplog.Info("unable to deserialize option 43")
+
+		return
+	}
```

Comment:
seems weird to do this check even when we know there was no previous Opt43. Why not move the Deserialize into an `else` branch of the above `if`? Actually I don't see the poing of doing anything at all when there is no pre-existing opt43 set, doesn't this work the same with less code?:

```go
n := make(dhcp4.OptionMap, 0)
if cur, ok := c.opts.GetOption(dhcp4.OptionVendorSpecific); ok {
	if err := n.Deserialize(cur, nil); err != nil {
		dhcplog.Info("unable to deserialize option 43")

		return
	}
}
```

*Actually*, after messing with similar code in `dhcp/pxe.go` why not just directly use the `OptionMap`?:

```go
n := make(dhcp4.OptionMap, 0)

// get current option 43
if cur, ok := c.opts.GetOption(dhcp4.OptionVendorSpecific); ok {
	if err := n.Deserialize(cur, nil); err != nil {
		dhcplog.Info("unable to deserialize option 43")

		return
	}
}

n[subOpt] = []byte(s)
```